mautic/mautic
# Frequency rule

Please answer the following questions. 

| Q  | A
| --- | ---
| Bug fix? | n
| New feature? | y
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/75
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | n/a
| BC breaks? | n
| Deprecations? | n

**Note that all new features should have a related user and/or developer documentation PR in their respective repositories.** 

### Required
#### Description:
This PR adds contact frequency rules to contacts to be applied at this stage to sms and email channels. This rule is applied per contact, and it sets the amount of times a contact should be contacted within a time frame.

#### Steps to test this PR:
1. Apply this PR
2. Go to any lead and click on it so that you are on the details page, in the dropdown arrow for the action buttons you should now see a new menu item to set the contact's frequency rules.
3. Once applied the contact should not be SMSed or emailed more than their set frequency rules. If a contact has no frequency rules applied it will send sms or emails as usual.
<img width="699" alt="screen shot 2016-07-26 at 22 45 11" src="https://cloud.githubusercontent.com/assets/1496976/17156613/ac515ae4-5382-11e6-8b8a-e8aece04989c.png">
